src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs

```diff
@@ -47,7 +58,7 @@ public async Task StreamsAreNotPersistedAcrossRequests()
                 context.Response.Body = new MemoryStream();
 
                 await context.Response.WriteAsync("hello, world");
-            }, new TestServiceContext(LoggerFactory)))
+            }, new TestServiceContext(LoggerFactory, _enableLineFeedTerminator)))
```

Comment:
For manual testing, I suggest also running the all the `RequestTests` with `\n` instead of `\r\n` as the line seperator in `StreamBackedTestConnection.Send` and `SendAll`:

https://github.com/dotnet/aspnetcore/blob/8f8be8eb4d87e22e5e622832a69da3cb527a8210/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs#L77-L79

https://github.com/dotnet/aspnetcore/blob/8f8be8eb4d87e22e5e622832a69da3cb527a8210/src/Servers/Kestrel/shared/test/StreamBackedTestConnection.cs#L68-L70

This would have demonstrated the issue with headers ending with `\n\n` instead of `\n\r\n`. The only false negative when you do this is the `ServerConsumesKeepAliveChunkedRequest` test because it uses the automatic joining of lines with "\r\n" for chunked request bodies, not just headers.

It is possible to flow this through the `TestServiceContext` if we really want to, but again I don't think we should have variations of all these tests checked in.